source/docs/essentials/create-a-ui-component.md

```diff
@@ -0,0 +1,307 @@
+  sometimes, but it will greatly improve the parent component's readability. For
+  instance, a UI component should not give the opportunity to pass a `className`
+  in its props as it may lead to many styles inconsistencies across the theme.
+
+## How to build a UI Component?
+
+Alright, that's nice in theory, but how does it translate in practice? We'll try
+to get a bit more tangible by creating a UI component needed for adding a
+Reinsurance Banner in a page.
+
+![The reinsurance banner that we will implement](assets/reinsurance.jpg)
+
+### Step 1: Defining the components
+
+First, let's split the mockup in several UI components.
+
+![The various components that will make up our banner](assets/reinsurance-with-areas.jpg)
+
+- **`theme/components/atoms/typography/Heading`:** enforces consistent font sizes in our theme
+  for any title
+- **`theme/components/atoms/Icon`:** enforces icon sizes and accessibility guidelines
+- **`theme/components/molecules/IllustratedContent`:** displays some content illustrated by an
+  image and aligns images consistently across the whole theme
+- **`theme/components/organisms/InlineCards`:** manages a list of cards and inlines them,
+  regardless of the device size.
+
+<blockquote class="note">
+If you have trouble splitting your mockups, you can refer to
+[Thinking in React](https://reactjs.org/docs/thinking-in-react.html) in the
+official React documentation or to
+[Brad Frost's book about Atomic Design](http://atomicdesign.bradfrost.com/).
+You may want to organize your code differently and that's perfectly fine. The
+way we splitted things here is one of many possible solutions. Such choices
+will depend on your project and your team. It may be a better idea to keep
+things simple. It's often easier to wait and see how it can be reused later.
+</blockquote>
+
+We won't be able to detail each component here. We will focus on
+`theme/components/molecules/IllustratedContent` instead. But keep in mind that
+any UI component in Front-Commerce will look similar to what we are going to
+build here.
+
+### Step 2: Setup your dev environment
+
+Before doing the actual work let's bootstrap our dev environment. To do so,
+once you've [registered your module](extend-the-theme.html#Configure-your-custom-theme-and-use-it-in-your-application) you will need to create three files:
+
+- `my-module/web/theme/components/molecules/IllustratedContent/IllustratedContent.js`: will be
+  your actual component
+
+  ```jsx
+  import React from "react";
+
+  const IllustratedContent = () => <div>Illustrated Content</div>;
+
+  IllustratedContent.propTypes = {
+    // Don't forget to setup your PropTypes
+    // here since this component will be heavily
+    // used through your application
+  };
+
+  export default IllustratedContent;
+  ```
+
+- `my-module/web/theme/components/molecules/IllustratedContent/index.js`: will proxy the
+  IllustratedContent.js file in order to be able to do imports on the folder
+  directly.
```

Comment:
```suggestion
  directly. See [this blog post](http://bradfrost.com/blog/post/this-or-that-component-names-index-js-or-component-js/) for more context about this pattern.
```

It could be a blockquote too if it seems better.